modules/sdk-coin-sui/src/sui.ts

```diff
@@ -92,8 +95,31 @@ export class Sui extends BaseCoin {
     return true;
   }
 
-  isWalletAddress(params: VerifyAddressOptions): boolean {
-    throw new Error('Method not implemented.');
+  async isWalletAddress(params: TssVerifyAddressOptions): Promise<boolean> {
+    const { keychains, address: newAddress, index } = params;
+
+    if (!this.isValidAddress(newAddress)) {
+      throw new InvalidAddressError(`invalid address: ${newAddress}`);
+    }
+
+    if (!keychains) {
+      throw new Error('missing required param keychains');
+    }
+
+    for (const keychain of keychains) {
+      const MPC = await EDDSAMethods.getInitializedMpcInstance();
+      const commonKeychain = keychain.commonKeychain as string;
+
+      const derivationPath = 'm/' + index;
```

Comment:
i think this is going to break for custodial wallets, where the derivation path is randomly generated to look something like:
`m/{rand1}/{rand2}/{addressIndex}`

Reply:
Created this ticket to come back to this
BG-59928